src/RAM/Science/components/Fluorometer.tsx

```diff
@@ -0,0 +1,215 @@
+  marginTop: "5px",
+  marginBottom: "5px",
+}
+const button: CSS.Properties = {
+  marginLeft: "15px",
+  width: "60px",
+}
+
+const controlButton: CSS.Properties = {
+  //margin: "5px",
+}
+
+/** Will be merged with the row css if the Laser is off */
+const offIndicator: CSS.Properties = {
+  backgroundColor: "#FF0000",
+}
+
+/** Will be merged with the row css if the Laser is off */
+const onIndicator: CSS.Properties = {
+  backgroundColor: "#00FF00",
+}
+
+interface IProps {
+  style?: CSS.Properties
+}
+
+interface IState {
+  /** Holds the last sent values of the diodes */
+  DiodeValues: number[]
+  /** Holds which lasers are enabled */
+  LasersPowered: boolean[]
+  /** If UV light is on */
+  UVPowered: boolean
+  /** If White Light is on */
+  WhiteLightPowered: boolean
+}
+
+class Fluorometer extends Component<IProps, IState> {
+  constructor(props: IProps) {
+    super(props)
+    this.state = {
+      DiodeValues: [0, 0, 0],
+      LasersPowered: [false, false, false],
+      UVPowered: false,
+      WhiteLightPowered: false,
+    }
+    this.updateDiodeVals = this.updateDiodeVals.bind(this)
+    this.toggleWhiteLight = this.toggleWhiteLight.bind(this)
+    this.toggleUV = this.toggleUV.bind(this)
+    this.buildLightCommand = this.buildLightCommand.bind(this)
+    this.buildLaserCommand = this.buildLaserCommand.bind(this)
+    this.exportData = this.exportData.bind(this)
+
+    rovecomm.on("FluorometerData", (data: any) => this.updateDiodeVals(data))
+  }
+
+  /**
+   * Updates the wavelengths received from the Rover.
+   * @param data float array of length 3 with the new data
+   */
+  updateDiodeVals(data: number[]): void {
+    const { DiodeValues } = this.state
+    DiodeValues[0] = data[0]
+    DiodeValues[1] = data[1]
+    DiodeValues[2] = data[2]
+    this.setState({ DiodeValues })
+  }
+
+  buildLightCommand(UV: boolean, White: boolean): number {
+    let bitmask = ""
+    bitmask += UV ? "1" : "0"
+    bitmask += White ? "1" : "0"
+    return parseInt(bitmask, 2)
+  }
+
+  buildLaserCommand(Lasers: boolean[]): number {
+    let bitmask = ""
+    bitmask += Lasers[0] ? "1" : "0"
+    bitmask += Lasers[1] ? "1" : "0"
+    bitmask += Lasers[2] ? "1" : "0"
+    return parseInt(bitmask, 2)
+  }
+
+  toggleWhiteLight(): void {
+    let { WhiteLightPowered } = this.state
+    WhiteLightPowered = !WhiteLightPowered
+    rovecomm.sendCommand("Lights", [this.buildLightCommand(this.state.UVPowered, WhiteLightPowered)])
+    this.setState({ WhiteLightPowered })
+  }
+
+  toggleUV(): void {
+    let { UVPowered } = this.state
+    UVPowered = !UVPowered
+    rovecomm.sendCommand("Lights", [this.buildLightCommand(UVPowered, this.state.WhiteLightPowered)])
+    this.setState({ UVPowered })
+  }
+
+  toggleLaser(index: number): void {
+    const { LasersPowered } = this.state
+    LasersPowered[index] = !LasersPowered[index]
+    rovecomm.sendCommand("FLasers", [this.buildLaserCommand(LasersPowered)])
+    this.setState({ LasersPowered })
```

Comment:
I suggest doing what I suggested with `toggleWhiteLight()` in these functions as well, or just making all three toggles one-lines in the html